src/Frame.cc

```diff
@@ -467,15 +507,27 @@ Frame::UnserializeIDList(const broker::vector& data)
 		{
 		auto has_name = broker::get_if<std::string>(*where);
 		if ( ! has_name )
+			{
+			for ( auto& i : rval )
+				Unref(i);
+
+			rval = id_list{};
 			return std::make_pair(false, std::move(rval));
+			}
 
 		ID* id = new ID(has_name->c_str(), SCOPE_FUNCTION, false);
 
 		std::advance(where, 1);
 
 		auto has_offset = broker::get_if<broker::integer>(*where);
 		if ( ! has_offset )
+			{
+			for ( auto& i : rval )
+				Unref(i);
+
+			rval = id_list{};
```

Comment:
Doesn't this still leak the `id` new'd above?

Reply:
Yep, fixed it, thanks.